src/test/system.rs

```diff
@@ -179,6 +183,16 @@ fn run_sccache_command_test(sccache: &Path, compiler: Compiler, tempdir: &Path)
         let expected_lines: Vec<_> = expected.lines().collect();
         assert_eq!(lines, expected_lines);
     }
+    if name == "gcc" {
+        trace!("test -MP with -Werror");
```

Comment:
I think it would be better to do something less relying on multiple things going wrong at the same time, which this test does, and instead, test the actual underlying problem.

So, something like, put a warning producing code in a #ifdef FOO, and pass -Wall -Werror -DFOO=1 on the command line.